main.js

```diff
@@ -0,0 +1,910 @@
+                        })
+                    })
+                ])
+                if(checkTorrent){
+                    checkTorrent.folder = folderPath
+                    let checkProperty = await Promise.any([
+                        new Promise((resolve, reject) => {setTimeout(() => {resolve(null)}, self._status.timeout)}),
+                        new Promise((resolve, reject) => {
+                            self.ownData(checkInternal[i], checkTorrent.infoHash).then(res => {
+                                resolve(res)
+                            }).catch(error => {
+                                console.log(error)
+                                // most likely the infohash of this torrent does not match what we have currently, stop this torrent
+                                self.webtorrent.remove(checkTorrent.infoHash, {destroyStore: false})
+                                resolve(null)
+                            })
+                        })
+                    ])
+                    if(checkProperty){
+                        delete checkProperty.infoHash
+                        for(const prop in checkProperty){
+                            checkTorrent[prop] = checkProperty[prop]
+                        }
+                        checkTorrent.folder = folderPath
+                        checkTorrent.side = true
+                        console.log(checkInternal[i] + ' is good')
+                    }
+                }
+            } else if(checkTitle.test(checkInternal[i])){
+                let checkTorrent = await Promise.any([
+                    new Promise((resolve, reject) => {
+                        setTimeout(() => {resolve(null)}, self._status.timeout)
+                    }),
+                    new Promise((resolve, reject) => {
+                        self.webtorrent.seed(folderPath, {destroyStoreOnDestroy: true}, torrent => {
+                            resolve(torrent)
+                        })
+                    })
+                ])
+                if(checkTorrent){
+                    checkTorrent.folder = folderPath
+                    checkTorrent.title = checkInternal[i]
+                    checkTorrent.side = true
+                    console.log(checkInternal[i] + ' is good')
+                }
+            } else {
+                await fs.remove(folderPath)
+            }
+        }
+    }
+
+    // if share option is true, then start seeding(webtorrent.add seeds after torrent is downloaded) on start up
+    if(self._status.share){
+        let checkExternal = await fs.readdir(self._external, {withFileTypes: false})
+        for(let i = 0;i < checkExternal.length;i++){
+            let folderPath = self._external + path.sep + checkExternal[i]
+            if(checkAddress.test(checkExternal[i])){
+                let checkProperty = await Promise.any([
+                    new Promise((resolve, reject) => {setTimeout(() => {resolve(null)}, self._status.timeout)}),
```

Comment:
i will work on this

Reply:
do you think it would be good to import a module just for the delay feature since we would be bringing in another module for only 1 thing?

Reply:
I mean, I usually just use the `delay` module, but defining it as a utility function is super easy too.

```
async function delay(time) {
  return new Promise((resolve)=> setTimeout(resolve, time))
}
```

Reply:
made the `delayTimeOut()` function, will that satisfy what we are looking for?

Reply:
Should be good! Might be useful to set a default for the last argument (e.g. instead of `delayTimeOut(timeout, data, res)` have `delayTimeOut(timeout, data, shouldResolve = false)`